### ISSUE TITLE
disable textInput not working if output: flexdashboard::flex_dashboard

### ISSUE THREAD

**Issue by bridgeportbyron** [daattali/shinyjs#67]
I am building Flexdashboard UI though R markdown. I am seeing that Disable textInput works if I remove `output: flexdashboard::flex_dashboard` in the title block.

```
         ---
        title: "without ui and server"
        runtime: shiny
        ---

         ```{r setup, include=FALSE}
          library(flexdashboard)
          library(shiny)
          library(shinyjs)
         ```

         Inputs {.sidebar}
         -----------------------------------------------------------------------

         ```{r, echo=FALSE}

          useShinyjs(rmd = TRUE)
          disabled(textInput("id", "Id", "0"))
          textInput("X1", "X1", "")

          #action buttons
          actionButton("submit", "Submit Changes")

          ```
```

If I include the line `output: flexdashboard::flex_dashboard`  it does not work. The textbox is not disabled.

```
                     ---
                     title: "without ui and server"
                     output: 
                     flexdashboard::flex_dashboard:
                     runtime: shiny
                     ---

                     ```{r setup, include=FALSE}
                        library(flexdashboard)
                        library(shiny)
                        library(shinyjs)
                      ```

                      Inputs {.sidebar}
                      -----------------------------------------------------------------------

                      ```{r, echo=FALSE}

                       useShinyjs(rmd = TRUE)
                       disabled(textInput("id", "Id", "0"))
                       textInput("X1", "X1", "")

                       #action buttons
                       actionButton("submit", "Submit Changes")

                      ```
```

**Comment by daattali** [comment]
I wasn't aware of the `flexdashboard` package until now. I just took a quick look and from what I can gather, this is actually a problem (bug/feature?) in flexdashboard rather than in shinyjs.  It seems that sending a javascript message using the typical method of doing so in shiny is broken.  The following code works in an Rmarkdown document:

```{r, echo=FALSE}
tags$script("Shiny.addCustomMessageHandler('mymsg', function(msg){ alert(msg); })")
textInput("text", "Message", "hello")
actionButton("submit", "Submit")

observeEvent(input$submit, {
  session$sendCustomMessage('mymsg', input$text)
})
```

if it has `runtime: shiny` in the YAML. But it does not work if the output is set to flex_dashboard. Also, I noticed that when the output is set to flex_dashboard, the "Run Document" button is changed to "knit". I'm not sure if that matters.  Perhaps I just don't know how to call JavaScript properly in this new type of rmd documents; I'll file an issue with them and see
